tests/test_cmp.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE FLOAT_CMP_TESTS
+#include <boost/test/unit_test.hpp>
+
+#include <stdexcept>
+
+#include <opm/common/util/numeric/cmp.hpp>
+
+using namespace Opm;
+
+/**
+   Ahhh - the joys of comparing floating point numbers ....
+
+   http://www.cygnus-software.com/papers/comparingfloats/comparingfloats.htm
+*/
+
+BOOST_AUTO_TEST_CASE(TestSCalarcmp) {
+    const double abs_epsilon = cmp::abs_epsilon;
+    const double rel_epsilon = cmp::rel_epsilon;
+
+    BOOST_CHECK( cmp::double_equal(1,1));
+    BOOST_CHECK_EQUAL( false , cmp::double_equal(1,0));
+    BOOST_CHECK_EQUAL( false , cmp::double_equal(0,1));
+    BOOST_CHECK_EQUAL( false , cmp::double_equal(-1,1));
+
+
+
+    double v1,v2;
+    /* Should be equal: */
+    {
+        v1 = 0.0;
+        v2 = 0.0;
+        BOOST_CHECK( cmp::double_equal( v1 , v2));
+
+        v1 = 1e-12;
+        v2 = v1 + 0.5*abs_epsilon;
+        BOOST_CHECK( cmp::double_equal( v1 , v2));
+
+        v1 = 7.0;
+        v2 = 7.0;
+        BOOST_CHECK( cmp::double_equal( v1 , v2));
+
+        v1 = -7.0;
+        v2 = -7.0;
+        BOOST_CHECK( cmp::double_equal( v1 , v2));
+
+        v1 = 0;
+        v2 = 0.5 * abs_epsilon;
+        BOOST_CHECK( cmp::double_equal( v1 , v2));
+
+
+        v1 = 1e7;
+        v2 = 1e7 + 2*abs_epsilon;
+        BOOST_CHECK( cmp::double_equal( v1 , v2 ));
+
+        v1 = 1e7;
+        v2 = 1e7*(1 + 2*rel_epsilon);
+        BOOST_CHECK( cmp::double_equal( v1 , v2 ));
```

Comment:
Thanks, I see it now. I think this is misleading though, and it would be better if this failed (as it will if we replace the sum with the largest absolute value).